niceman/utils.py

```diff
@@ -895,4 +896,21 @@ def unicode(s):
     else:
         return __builtin__.unicode(s)
 
+
+def generate_unique_name(pattern, nameset):
+    """Create a unique numbered name from a pattern and a set"""
+    i = 0
+    while True:
+        n = pattern % i
+        i += 1
+        if n not in nameset:
+            return n
+
+
+# http://stackoverflow.com/questions/1151658/python-hashable-dicts
+class HashableDict(dict):
+    """Dict that can be used as keys"""
+    def __hash__(self):
+        return hash(frozenset(self.values()))
```

Comment:
I need to confess that I worked on some code related to your work - identifying version control repos... Not yet functioning but we better merge this code and then continue improving in the next PRS... OK to merge? ;-)